tock/api/views.py

```diff
@@ -260,14 +307,14 @@ def filter_timecards(queryset, params={}):
 
     if 'after' in params:
         # get everything after a specified date
-        after_date = params.get('after')
+        after_date = date_from_iso_format(params.get('after'))
         queryset = queryset.filter(
             reporting_period__end_date__gte=after_date
         )
 
     if 'before' in params:
         # get everything before a specified date
-        before_date = params.get('before')
+        before_date = date_from_iso_format(params.get('before'))
```

Comment:
@Jkrzy and @neilmb not sure if this came up before, but I'm also wondering how we think about reporting periods in the context of the "before" and "after" arguments here - say we had a reporting period of April 1 - April 15. If someone queries for "before April 7", are they expecting to get timecards whose end date is _after_ April 7? Or should the entirety of the reporting period fall before April 7?

As-is now, if someone asks for timecards "before April 7", we would include ones from the time period April 1 - April 15 (because its start date is April 1, which falls before April 7). I could see it either way, but think it might be confusing the way it is now since you'd be including data from work that could've been done after April 7 (e.g. if someone tocked 10 hours on April 9, that would be counted).

Reply:
It looks to me like `after` gets every period whose end_date is after the specified date https://github.com/18F/tock/blob/main/tock/api/views.py#L265 and `before` gets every period whose start_date is before the specified date https://github.com/18F/tock/blob/main/tock/api/views.py#L272.

I think that I chose that way on purpose, giving the larger returned set of data so that you don't have to move a date forward (or backward) in time until you get past the end (start) date to get a time period to be returned.  (Hmm, was April 15, 2020 a Saturday?)

But yes, I remember thinking that there were two different ways to interpret this.

Reply:
Yes, agree with how the `before` and `after` params are working now! Just wanted to clarify/double check that that's how users of the API were using / understanding it. Thanks @neilmb!